plugins/module_utils/_global_catalog.py

```diff
@@ -67,64 +55,137 @@ def _get_catalog_records(stdout):  # type: (str) -> tuple[str | None, str | None
 
     return (autostart_override, nextstart)
 
+def _create_dfhrmutl_jcl(location, sdfhload, cmd=""):
+    steplib_line = f"//STEPLIB  DD DSNAME={sdfhload},DISP=SHR"
+    dfhgcd_line = f"//DFHGCD   DD DSNAME={location},DISP=OLD"
+    # Validate line lengths
+    _validate_line_length(steplib_line, sdfhload)
+    _validate_line_length(dfhgcd_line, location)
+    
+    jcl = ""
+    if (cmd ==""):
+        jcl = f'''
+//DFHRMUTL JOB
+//RMUTL    EXEC PGM=DFHRMUTL,REGION=1M
+{steplib_line}
+//SYSPRINT DD SYSOUT=*
+{dfhgcd_line}
+//SYSIN    DD *
+/*
+'''
+    else:
+        jcl = f'''
+//DFHRMUTL JOB
+//RMUTL    EXEC PGM=DFHRMUTL,REGION=1M
+{steplib_line}
+//SYSPRINT DD SYSOUT=*
+{dfhgcd_line}
+//SYSIN    DD *
+    {cmd}
+/*
+'''
+
+    # Create a temporary file
+    with tempfile.NamedTemporaryFile(mode='w+', delete=False) as fp:
+        fp.write(jcl)
+        fp.flush()
+
+        # Get the temporary file's name
+        qualified_file_path = fp.name
+    return qualified_file_path
+
+def _validate_line_length(line, name):
+    """
+    Validates that the JCL line does not exceed MAX_LINE_LENGTH.
+    Raises ValueError if validation fails.
+    """
+    if len(line) > MAX_LINE_LENGTH:
+        raise ValueError(f"{name} line exceeds {MAX_LINE_LENGTH} characters: {len(line)}")
+    return True
+
+def _validate_name_params(param):
+    if any(len(part) > MAX_NAME_LENGTH for part in param.split('.')):
```

Comment:
Is this needed? We do upfront checking with better arg parser that both SDFHLOAD and DFHGCD dsn would go through and that would do the 8 char check.